test-cypress/integration/repository/repositories.spec.js

```diff
@@ -0,0 +1,404 @@
+        // Note: the first one should be cleared in afterEach() preventing other tests to select it by default
+        getRepositoryFromList(repositoryId)
+            .find('.pin-repository-btn')
+            .should('not.have.class', 'active')
+            .click()
+            .should('have.class', 'active');
+        // The currently selected repository is kept as a cookie
+        cy.clearCookies();
+        cy.visit('/repository');
+
+        // Should automatically select the default repository
+        getRepositoriesDropdown()
+            .find('.active-repository')
+            .should('contain', repositoryId);
+
+        // TODO: Could push the repo in an array and afterEach can delete all of them
+        cy.deleteRepository(secondRepoId);
+    });
+
+    it('should allow to edit existing repository', () => {
+        const newTitle = 'Title edit';
+        const newBaseUrl = 'http://example.org/wine#';
+
+        createRepository();
+        typeRepositoryId(repositoryId);
+        typeRepositoryTitle('Title');
+        saveRepository();
+
+        editRepository(repositoryId);
+
+        // Some fields should be disabled
+        getRepositoryRulesetMenu().should('be.disabled');
+        getRepositoryDisableSameAsCheckbox().should('be.disabled');
+
+        typeRepositoryTitle(newTitle);
+        typeRepositoryBaseURL(newBaseUrl);
+        getRepositoryContextIndexCheckbox().check();
+
+        getSaveRepositoryButton().click();
+        confirmModal();
+        waitLoader();
+
+        // See the title is rendered in the repositories list
+        getRepositoryFromList(repositoryId).should('contain', newTitle);
+
+        // Go to edit again to verify changes
+        editRepository(repositoryId);
+
+        getRepositoryTitleField().should('have.value', newTitle);
+        getRepositoryBaseURLField().should('have.value', newBaseUrl);
+        getRepositoryContextIndexCheckbox().should('have.value', 'true');
+    });
+
+    it('should allow to delete existing repository', () => {
+        createRepository();
+        typeRepositoryId(repositoryId);
+        saveRepository();
+
+        selectRepoFromDropdown(repositoryId);
+
+        getRepositoryFromList(repositoryId)
+            .find('.repository-actions .delete-repository')
+            // Forcefully clicking it due to https://github.com/cypress-io/cypress/issues/695
+            .should('be.visible')
+            .and('not.be.disabled')
+            .click({force: true});
+
+        confirmModal();
+
+        getRepositoriesList().should('not.contain', repositoryId);
+
+        // Check the repo has been deselected and is not present in the repo dropdown menu
+        getRepositoriesDropdown().click().within(() => {
+            cy.get('.no-selected-repository').should('be.visible');
+            cy.get('.dropdown-menu .dropdown-item').should('not.contain', repositoryId);
+        });
+    });
+
+    const REPO_LIST_ID = '#wb-repositories-repositoryInGetRepositories';
+
```

Comment:
You tend to put these functions at the end of the spec. Maybe I'll start placing mine at the end as well.

Reply:
Well.. I think it's more important to see the spec and what is tested/verified instead of reading some functions with selections, actions etc.